### PR TITLE
ARROW-8044: [CI][NIGHTLY:gandiva-jar-osx] Pin pygit2 at 1.0.3 for OSX

### PR DESCRIPTION
**Pull request by bkietz**

### CODE REVIEW

**Comment by bkietz** [comment]
@ursabot crossbow -g gandiva-jar-osx

**Comment by projjal** [comment]
Doing pip3 install pygit2==1.0.3 (the earlier version) here https://github.com/apache/arrow/blob/aec37d768abbee3ea3ce4c32002462ff0e1c3674/dev/tasks/gandiva-jars/travis.osx.yml#L60 should solve the issue. The brew libgit2 package is not yet updated to 1.0.x

**Comment by bkietz** [comment]
@projjal thanks, I'll try that

File: dev/tasks/gandiva-jars/travis.osx.yml [hunk 1]
```diff
@@ -57,7 +57,8 @@ script:
   # gandiva libraries are already built we can reset MACOS_DEPLOYMENT_TARGET to current osx version
   - if [[ $MACOSX_DEPLOYMENT_TARGET ]] ; then export MACOSX_DEPLOYMENT_TARGET=$(sw_vers -productVersion | cut -d '.' -f 1,2) ; fi
   - brew install libgit2
-  - pip3 install click github3.py jinja2 jira pygit2 ruamel.yaml setuptools_scm toolz
+  - pip3 install pygit2=1.0.3
```
Comment (projjal) [inline dev/tasks/gandiva-jars/travis.osx.yml]:
```suggestion
  - pip3 install pygit2==1.0.3
```

**Comment by bkietz** [comment]
@ursabot crossbow --help

**Comment by bkietz** [comment]
Manual crossbow https://travis-ci.org/bkietz/crossbow/builds/660702789

**Review by wesm — APPROVED** [review]
+1. thanks @bkietz!

**Pull request closed without merging**